### ISSUE TITLE
add warning count to json report

### ISSUE THREAD

**Issue by djwgit** [cucumber/cucumber-ruby#930]
currently, a test feature could succeed or fail.  if fail, it stops and fails the test feature.  in the json report, it says counts of passed/failed steps...

in some case, like to have a count of warning in test result report,  when a warning is given at one step, it doesn't stop the test workflow.
say, 
- in the middle of a test workflow. verify a number (say, count of a list). the test finds that it is a wrong number. in this case. it is not a big showstopper, don't want to fail the test workflow.  just want to record it as a warning, continue to finish the rest of the test workflow.
- and the warning could be added to the json report. it shows how many warnings
- in console output,  nice to have a warning output in red text...

saw, there is a function called _warn_, but it is not being added to json report,
basically,  the test result status of a step could be "warning"

```
            "result": {
              "status": "passed",           // "warning"
              "duration": 1130953000
            }
```

just log this as an enhancement, might be already a solution there I don't know of...   thanks.

**Comment by danascheider** [comment]
@djwgit Sorry to take so long getting to this! I'm not sure I understand what you're trying to do. Are you looking for behavior similar to the RSpec [aggregate_failures](https://www.relishapp.com/rspec/rspec-core/docs/expectation-framework-integration/aggregating-failures#use-`aggregate-failures`-block-form) behavior? That is, do you want the entire scenario to run even when one step fails? Or did you have something else in mind?

**Comment by djwgit** [comment]
thanks @danascheider for your reply.  in some case, in a test step, the test code find an issue, but not critical (say, a text label message not spelled right...), want to log it as an warning, but still have the rest test steps continue to run.   
and use warn() function to print out a warning message in the output json report.
